packages/flutter_web_plugins/lib/src/navigation_non_web/url_strategy.dart

```diff
@@ -0,0 +1,217 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:ui' as ui;
+
+/// Signature of an html event listener.
+typedef EventListener = void Function(Object event);
+
+/// Change the strategy to use for handling browser URL.
+///
+/// Setting this to null disables all integration with the browser history.
+void setUrlStrategy(UrlStrategy? strategy) {
+  // No-op in non-web platforms.
+}
+
+/// Use the [PathUrlStrategy] to handle the browser URL.
+void usePathUrlStrategy() {
+  // No-op in non-web platforms.
+}
+
+/// Represents and reads route state from the browser's URL.
+///
+/// By default, the [HashUrlStrategy] subclass is used if the app doesn't
+/// specify one.
+abstract class UrlStrategy {
```

Comment:
I'd recommend to export the common classes from another file, and only duplicate the things that really need a different implementation.